internal/controller/clusterdeployment_reconciler_test.go

```diff
@@ -0,0 +1,370 @@
+				Conditions: []hivev1.ClusterDeploymentCondition{
+					{
+						Type:    hivev1.ClusterInstallFailedClusterDeploymentCondition,
+						Status:  corev1.ConditionStatus(metav1.ConditionFalse),
+						Reason:  "InstallationNotFailed",
+						Message: "The installation has not failed"},
+				},
+			},
+		}
+		Expect(c.Create(ctx, clusterDeployment)).To(Succeed())
+
+		res, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: key})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(ctrl.Result{}))
+
+		// Fetch clusterdeployment and verify that the status is unchanged
+		cd := &hivev1.ClusterDeployment{}
+		Expect(c.Get(ctx, key, cd)).To(Succeed())
+		Expect(cd.Spec).To(Equal(clusterDeployment.Spec))
+		Expect(cd.Status).To(Equal(clusterDeployment.Status))
+	})
+
+	It("tests that ClusterDeploymentReconciler initializes SiteConfig ClusterDeployment correctly", func() {
+		key := types.NamespacedName{
+			Namespace: clusterNamespace,
+			Name:      clusterName,
+		}
+		clusterDeployment := &hivev1.ClusterDeployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      clusterName,
+				Namespace: clusterNamespace,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: "metaclusterinstall.openshift.io/v1alpha1",
+						Kind:       v1alpha1.Kind,
+						Name:       clusterName,
+					},
+				},
+			},
+		}
+		Expect(c.Create(ctx, clusterDeployment)).To(Succeed())
+
+		res, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: key})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(res).To(Equal(requeueImmediately()))
+
+		expectedConditions := []hivev1.ClusterDeploymentCondition{
+			{
+				Type:    hivev1.ClusterInstallRequirementsMetClusterDeploymentCondition,
+				Status:  corev1.ConditionStatus(metav1.ConditionUnknown),
+				Message: "Unknown",
+			},
+			{
+				Type:    hivev1.ClusterInstallStoppedClusterDeploymentCondition,
+				Status:  corev1.ConditionStatus(metav1.ConditionUnknown),
+				Message: "Unknown",
+			},
+			{
+				Type:    hivev1.ClusterInstallCompletedClusterDeploymentCondition,
+				Status:  corev1.ConditionStatus(metav1.ConditionUnknown),
+				Message: "Unknown",
+			},
+			{
+				Type:    hivev1.ClusterInstallFailedClusterDeploymentCondition,
+				Status:  corev1.ConditionStatus(metav1.ConditionUnknown),
+				Message: "Unknown",
+			},
+		}
+
+		sc := &v1alpha1.SiteConfig{}
+		Expect(c.Get(ctx, key, sc)).To(Succeed())
+		Expect(sc.Status.ClusterDeploymentRef.Name).To(Equal(clusterName))
+		Expect(len(sc.Status.DeploymentConditions)).To(Equal(len(expectedConditions)))
+
+		matchedConditions := true
+		for _, cond := range expectedConditions {
+			matched := false
+			for i := range sc.Status.DeploymentConditions {
+				if sc.Status.DeploymentConditions[i].Type == cond.Type &&
+					sc.Status.DeploymentConditions[i].Status == cond.Status {
+					matched = true
+				}
+			}
+			matchedConditions = (matchedConditions && matched)
+		}
+		Expect(matchedConditions).To(Equal(true))
```

Comment:
I think it will be easier here to assert in the loop, then you save one tracking bool.

```suggestion
		for _, cond := range expectedConditions {
			matched := false
			for i := range sc.Status.DeploymentConditions {
				if sc.Status.DeploymentConditions[i].Type == cond.Type {
					Expect(sc.Status.DeploymentConditions[i].Status).To(Equal(cond.Status))
					matched = true
				}
			}
			Expected(matched).To(Equal(true), "Condition %s was not found", cond.Type)
		}
```